inst/tests/tests.Rraw

```diff
@@ -16846,3 +16846,8 @@ A = data.table(A=c(complex(real = 1:3, imaginary=c(0, -1, 1)), NaN))
 test(2138.3, rbind(A,B), data.table(A=c(as.character(A$A), B$A)))
 A = data.table(A=as.complex(rep(NA, 5)))
 test(2138.4, rbind(A,B), data.table(A=c(as.character(A$A), B$A)))
+
+# compatible branches might seem incompatible if the condition is global: #4274
+DT = data.table(a=1L)
+test(2139, DT[ , if (TRUE) .(a=1) else .(a=1, b=2)], DT,
```

Comment:
`else` branch is ever evaluated?

Reply:
we evaluate its names, and the logic is to take the "last" names from j; in this case, it'll be the else branch.

in the test example it's obvious the second branch is spurious, but in general it's hard to tell lazily which branches will be evaluated